### ISSUE TITLE
Media upload button is not rendering

### ISSUE THREAD

**Issue by jvisick** [WordPress/gutenberg#5860]
## Issue Overview

Working locally off of current master #f4c40f6 it looks like the MediaUpload component is not rendering anymore? I don't see it on the image block or the cover image block, as well as in the inspector for featured images.

## Steps to Reproduce (for bugs)

1. Open new page
2. Add an image block
3. Only option to upload new image is given

## Screenshots / Video

![screen shot 2018-03-28 at 2 23 55 pm](https://user-images.githubusercontent.com/2252216/38057448-5f394206-3294-11e8-952d-bf893332787e.png)

![screen shot 2018-03-28 at 2 33 49 pm](https://user-images.githubusercontent.com/2252216/38057698-19bad202-3295-11e8-868e-1e32e3011db1.png)

**Comment by youknowriad** [comment]
I can't reproduce this myself. It seems to work as expected for me.

**Comment by danielbachhuber** [comment]
@jvisick What debugging have you done? Can you see if there are any JavaScript errors in your console?

**Comment by jvisick** [comment]
@danielbachhuber No console errors. I have tried to isolate what commit it still works on. @youknowriad if you can't recreate it let me try and get some more info...

**Comment by jvisick** [comment]
@danielbachhuber Okay, it appears to be a plugin conflict with plugin ACF Pro (v5.6.9). I'll see if I can isolate it further.

**Comment by youknowriad** [comment]
pinging @gziolo as this is probably related to the work we did on adding extensibility to the MediaUpload component #5211

**Comment by jvisick** [comment]
@youknowriad @gziolo yes the merge of #5211 seems to be where the conflict was introduced.

**Comment by gziolo** [comment]
I couldn't reproduce with the free version of AFC:

![screen shot 2018-03-29 at 12 44 36](https://user-images.githubusercontent.com/699132/38084927-0184ef46-334f-11e8-93c1-e1928f6fcd72.png)

Does it need to be a pro version? Do you have any specific fields configured which might cause some conflicts?

**Comment by shoetten** [comment]
I experience the same issue. It's not necessarily bound to the pro version, just to version 5 i think.
You can install ACF v5 either through composer or using the instructions described here:
https://www.advancedcustomfields.com/resources/upgrade-guide-version-5/

**Comment by mejta** [comment]
Same for me ... using ACF Pro 5.6.9. When I disable the plugin it works, if I enable plugin, MediaUpload is not shown.

**Comment by jvisick** [comment]
@gziolo I tested with a clean install of wp and just activating ACF Pro (v5.6.9) without any config causes the problem. I have not tried the free or earlier versions of ACF.

**Comment by shoetten** [comment]
React incorrectly renders an `<Unkown>` component, instead of the actual button, when used together with ACF => v5.
![react-dom-gutenberg-acf-incorrect](https://user-images.githubusercontent.com/1785212/38106489-6db1cc8a-338f-11e8-9541-6fbb437fa3c3.png)

**Comment by gziolo** [comment]
I was able to reproduce by installing ACF Pro (v5.6.9). I also found that other hooks don't work as expected. `Copy Content` button is missing in `More Menu`:

![screen shot 2018-03-30 at 12 18 09](https://user-images.githubusercontent.com/699132/38134731-a7779960-3414-11e8-9ea4-4312758afc4d.png)

This issue is caused by the ACF implementation in `assets/js/acf-input.js` file, lines 8-250:
```js
var EventManager = function() {
    ...
};

window.wp = window.wp || {};
window.wp.hooks = new EventManager();
```

Putting this into words, ACF reimplements `wp.hooks` instead of using the already present module registered under `wp-hooks` name. My understanding is that it removes some of the registered previously filters causing the issue.

There is nothing to do about it on the Gutenberg side, so closing this one.

**Comment by gziolo** [comment]
Thanks for your help with debugging this one. Any hints what is the best way to reach out to ACF team and help them fix this issue?

**Comment by gziolo** [comment]
Okey, I opened an issue in ACF repository ☝️

**Comment by gziolo** [comment]
> This has been fixed in ACF PRO 5.6.10. When you have a chance, can you please update and test it out? If the problem persists, please let me know and I'll continue to investigate.

@jvisick, @mejta, @shoetten - can you confirm? :)

**Comment by mejta** [comment]
@gziolo I can confirm that issue with MediaUpload is fixed with ACF PRO 5.6.10. When activated it works as expected.

**Comment by gziolo** [comment]
I bet they need to updated the names of their hooks to follow whatever wp.hooks expects. It’s definitely fixable. Thanks for your help 👏
